desc/objectives/_coils.py

```diff
@@ -602,11 +603,157 @@ def compute(self, params, constants=None):
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str, optional
+        Name of the objective function.
+    """
+
+    _scalar = False
+    _units = "(m)"
+    _print_value_fmt = "Minimum coil-coil distance: {:10.3e} "
+
+    def __init__(
+        self,
+        coils,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        loss_function=None,
+        deriv_mode="auto",
+        grid=None,
+        name="coilset minimum distance",
+    ):
+        if target is None and bounds is None:
+            bounds = (0, np.inf)
+        self._coils = coils
+
+        super().__init__(
+            coils,
+            ["x"],
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            grid=grid,
+            name=name,
+        )
+
+    def build(self, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        from desc.coils import CoilSet, _Coil
+
+        super().build(use_jit=use_jit, verbose=verbose)
+
+        if self._normalize:
+            self._normalization = self._scales["a"]
+
+        # TODO: repeated code but maybe it's fine
+        flattened_coils = tree_leaves(
+            self._coils,
+            is_leaf=lambda x: isinstance(x, _Coil) and not isinstance(x, CoilSet),
+        )
+        flattened_coils = (
+            flattened_coils
+            if isinstance(self._coils, CoilSet)
+            else [flattened_coils[0]]  # XXX: I don't understand this
+        )
+        self._dim_f = len(flattened_coils)  # FIXME: this is overkill for getting dim_f
+        self._constants["quad_weights"] = 1
+
+    def compute(self, params, constants=None):
+        """Compute coil torsion.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of the coil's degrees of freedom.
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self._constants.
+
+        Returns
+        -------
+        f : array of floats
+            Distance to nearest coil for each coil in the coilset.
+
+        """
+        pts = self._coils.compute_position(source_grid=self._grid)
+
+        def body(i):
+            # FIXME: this is only element-wise distances! we need all permutations
+            dx = pts[i] - pts
+            dist = safenorm(dx, axis=-1)
```

Comment:
@f0uriest @dpanici I don't think this is actually computing what we want. This is the element-wise distances between the "same" grid points on each coil, i.e., the distances between nodes with the same index. But we also want to include distances between all nodes with other indices in each coil's grid. In other words, if there are 2 coils with 5 nodes along each coil, this will return 5 distances for each coil, but what we really want is 5^2=25 distances for each coil. 

Let me know if someone thinks of a way to fix this.

Reply:
I think I fixed this now

Reply:
did you have a test where the old way failed but the new way works? I am unsure if I follow what you are saying

Reply:
This is covered in `test_coil_min_distance` by the last case with a `MixedCoilset`. In all the other cases, the minimum points on each coil all happen to be at the same grid point (like all at the inboard midplane, for example)